sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/LiveMetricsExtensions.cs

```diff
@@ -45,29 +45,34 @@ public static class LiveMetricsExtensions
                     // configuration delegate into options pipeline.
                     services.Configure(finalOptionsName, configure);
                 }
-            });
-
-            return builder.AddProcessor(sp =>
-            {
-                // SETUP OPTIONS
-                LiveMetricsExporterOptions exporterOptions;
 
-                if (name == null)
+                // Register Manager as a singleton
+                services.AddSingleton<Manager>(sp =>
```

Comment:
Are these operations thread-safe?

https://github.com/Azure/azure-sdk-for-net/blob/d9ab692c7e07f433e436e6a723a6d159509340ec/sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/LiveMetricsActivityProcessor.cs#L114

Making this a singleton means `_documentBuffer.WriteDocument` needs to be safe to be called concurrently. Just double-checking.

Reply:
the implementation uses ConcurrentQueue, so I hope it's safe 😉

Reply:
correct, this is safe to be called concurrently.

Reply:
One more somewhat related comment. This bit...

https://github.com/Azure/azure-sdk-for-net/blob/d9ab692c7e07f433e436e6a723a6d159509340ec/sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/LiveMetricsActivityProcessor.cs#L36

Looks like the `TracerProvider` resource is used to set the `Manager` singleton `LiveMetricsResource`. How will that work with a log processor? Possible that tracing and logging have different resources.

Reply:
We are moving this code to distro and deprecating this package. In distro we have a single resource that is applied to all 3 signals.